src/libYARP_math/include/yarp/math/Vec2D.h

```diff
@@ -0,0 +1,163 @@
+/*
+* Author: Marco Randazzo
+* Copyright (C) 2017 iCubFacility - Istituto Italiano di Tecnologia
+* CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+*/
+
+#ifndef YARP_VEC2D
+#define YARP_VEC2D
+
+#include <yarp/math/api.h>
+#include <yarp/sig/Vector.h>
+#include <yarp/sig/Matrix.h>
+#include <yarp/os/Portable.h>
+#include <type_traits>
+
+/// network stuff
+#include <yarp/os/NetInt32.h>
+
+namespace yarp
+{
+    namespace math
+    { 
+        template <typename T>
+        class Vec2D;
+    }
+}
+
+template <typename T>
+class yarp::math::Vec2D : public yarp::os::Portable
+{
+    static_assert (std::is_same<int, T>::value ||
+                   std::is_same<double, T>::value, "Vec2D can be specialized only as int, double");
+public:
+    T x;
+    T y;
+
+public:
+    Vec2D<T>();
+    Vec2D<T>(T x_value, T y_value);
+    Vec2D<T>(yarp::sig::Vector v);
+
+    explicit operator yarp::sig::Vector() const
+    {
+        yarp::sig::Vector v(2);
+        v[0] = double(x);
+        v[1] = double(y);
+        return v;
+    }
+
+    /**
+    * Returns the modulus of the Vec2D, i.e. srqrt(x^2+y^2)
+    * @return the computed modulus
+    */
+    T mod() const;
```

Comment:
I am afraid that `mod` could be ambiguous as a term, check for example: https://it.mathworks.com/help/matlab/ref/mod.html . 
An alternative name for this method that is used both in Matlab, Eigen and YARP itself (http://www.yarp.it/namespaceyarp_1_1math.html#a378045a0a2b0cc5e7d9c6e54709ba127) is `norm`.